### ISSUE TITLE
shadow-su: passwd not applied

### ISSUE THREAD

**Issue by kerner1000** [Entware-ng/Entware-ng#594]
### Package: shadow-su

I installed `shadow-su` to get the `su` command.

```
[~] # whoami
admin
[~] # su alex
[/root] # whoami
whoami: unknown uid 1000
[/root] # cd
[~] # pwd
/share/homes/alex
```
It seems `/etc/passwd` is not read/ applied correctly, since the user is not recognized. Strangely, the `su` command itself works, and also the correct home directory is recognized. But only after login when executing `cd` without parameters.

```
# ls -la /opt/etc/passwd
lrwxrwxrwx    1 admin    administ        11 Mar  5 16:35 /opt/etc/passwd -> /etc/passwd
# ls -la /etc/passwd
lrwxrwxrwx    1 admin    administ        13 Feb 12 23:18 /etc/passwd -> config/passwd

```
### Platform
System is a QNAP, firmware 4.2.3.

**Comment by zyxmon** [comment]
It looks like you are using Entware-3x, not Entware-ng. Please start your next issue here in this case - https://github.com/Entware-for-kernel-3x/Entware-ng-3x/issues
shadow* packages have some problems, we aware about it.
Have you tried busybox su applet before installing shadow-su?

**Comment by kerner1000** [comment]
@zyxmon many thanks for helping out! I will try busybox su applet and/ or create a new issue at the correct place. 
Can you point me to some documentation on how to install the su applet? I was not aware of busybox applets.

**Comment by zyxmon** [comment]
If you are using Entware-3x it was already installed. You have overwritten it with shadows-su.
1. Uninstall `opkg remove shadows-su`
2. Restore applet by running `/opt/lib/opkg/info/busybox.postinst`

**Comment by kerner1000** [comment]
It seems I am missing this executable..
```
# ls /opt/lib/opkg/info/busy*
ls: /opt/lib/opkg/info/busy*: No such file or directory
```

**Comment by zyxmon** [comment]
Busybox is installeв with entware-3x automatically. In entware-ng manually `opkg install busybox`.

**Comment by kerner1000** [comment]
Busybox is the QNAP OS, so it should be there anyways.

Now that I have removed the shadow* packages, I am again missing the `su` command.. According to QNAP support, `su` is not part of the default Busybox installation.

**Comment by zyxmon** [comment]
>> Busybox is the QNAP OS, so it should be there anyways.

You can install Entware version of busybox with su applet and use both versions!

**Comment by kerner1000** [comment]
OK, did so.
But still the same problem:

```
/opt/bin/opkg install busybox
Installing busybox (1.25.1-2) to root...
Downloading http://pkg.entware.net/binaries/x86-64/busybox_1.25.1-2_x86-64.ipk
Configuring busybox.
Warning: '/opt/bin/xargs' is already exists, skipping.
Warning: '/opt/bin/rm' is already exists, skipping.
Warning: '/opt/bin/mv' is already exists, skipping.
Warning: '/opt/bin/find' is already exists, skipping.
Warning: '/opt/bin/cp' is already exists, skipping.
Warning: '/opt/bin/less' is already exists, skipping.
[~] # /opt/bin/opkg remove busybox
Refusing to remove essential package busybox.
	Removing an essential package may lead to an unusable system, but if
	you enjoy that kind of pain, you can force opkg to proceed against
	its will with the option: --force-removal-of-essential-packages
No packages removed.
[~] # su alex
[/root] # whoami
whoami: unknown uid 1000
```

**Comment by zyxmon** [comment]
```
which whoami
/usr/bin/whoami
```
May be `whoami` that is a part of Qnap firmware has bugs? If the user alex has uid 1000, then both su versions are working perfectly!

**Comment by zyxmon** [comment]
Althouth it is complete offtopic, this will help you. It is not QNAP

```
root@ws880:/tmp/home/root# cat /etc/passwd 
root:x:0:0:root:/root:/bin/sh
nas:x:100:100:nas:/dev/null:/dev/null
nobody:x:65534:65534:nobody:/dev/null:/dev/null
root@ws880:/tmp/home/root# opkg install coreutils-whoami
Installing coreutils-whoami (8.23-2) to root...
Downloading http://pkg.entware.net/binaries/armv7/coreutils-whoami_8.23-2_armv7soft.ipk
Installing coreutils (8.23-2) to root...
Downloading http://pkg.entware.net/binaries/armv7/coreutils_8.23-2_armv7soft.ipk
Configuring coreutils.
Configuring coreutils-whoami.
root@ws880:/tmp/home/root# su nas -p
nas@ws880:/tmp/home/root$ whoami
nas
```

**Comment by kerner1000** [comment]
Thanks again for helping out!

I did as you recommended.

```
/opt/bin/opkg install coreutils-whoami
Installing coreutils-whoami (8.23-2) to root...
Downloading http://pkg.entware.net/binaries/x86-64/coreutils-whoami_8.23-2_x86-64.ipk
Configuring coreutils-whoami.
```
Logged out and in again.
Then

```
[~] # su alex
[/root] # whoami
whoami: cannot find name for user ID 1000
[/root] # 
```
Yes, user alex has id 1000. But why can `whoami` not find the name for the user id?

**Comment by zyxmon** [comment]
Please continue your questions on qnap forum.
